lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -1112,13 +1139,13 @@ impl SegmentHolder {
                 log::error!("Failed to replicate proxy segment field indexes, ignoring: {err}");
             }
 
-            let (segment_id, segments) = write_segments.swap_new(proxy, &[original_segment_id]);
-            debug_assert_eq!(segments.len(), 1);
+            // We must keep existing segment IDs because ongoing optimizations might depend on the mapping being the same
+            write_segments.replace(segment_id, proxy)?;
```

Comment:
Critical part.

This introduced new segment IDs before, now it keeps the existing mapping.